cmd/kubectl-gadget/utils/trace.go

```diff
@@ -290,32 +466,159 @@ RetryLoop:
+		}
+
+		customResultsDisplay(nodes, &traces)
 	}
-	deleteTraces(contextLogger, traceRestClient, traceID)
+
+	return nil
+}
+
+// DeleteTrace deletes the traces for the given trace ID using RESTClient.
+func DeleteTrace(traceID string) error {
+	traceRestClient, err := getRestClient()
+	if err != nil {
+		return err
+	}
+
+	deleteTraces(traceRestClient, traceID)
+
+	return nil
+}
+
+// getTraceListFromParameters returns traces associated with the given config.
+func getTraceListFromParameters(config *TraceConfig) ([]gadgetv1alpha1.Trace, error) {
+	/*
+	 * If some of the arguments are empty, they will just be ignored when
+	 * searching for corresponding trace.
+	 */
+	labels := fmt.Sprintf("GadgetName=%v,nodeName=%v,namespace=%v,podName=%v,containerName=%v,outputMode=%v",
+		config.GadgetName, config.NodeName, config.NamespaceName, config.PodName, config.ContainerName, config.TraceOutputMode)
+
+	var listTracesOptions = metav1.ListOptions{
+		LabelSelector: labels,
+		FieldSelector: fields.Everything().String(),
+	}
+
+	traces, err := getTraceListFromOptions(listTracesOptions)
+	if err != nil {
+		return []gadgetv1alpha1.Trace{}, err
+	}
+
+	return traces.Items, nil
+}
+
+// PrintAllTraces prints all traces corresponding to the given config.
+func PrintAllTraces(config *TraceConfig) error {
+	traces, err := getTraceListFromParameters(config)
+	if err != nil {
+		return err
+	}
+
+	w := tabwriter.NewWriter(os.Stdout, 0, 0, 4, ' ', 0)
+
+	fmt.Fprintln(w, "NAMESPACE\tPOD\tCONTAINER\tTRACEID")
+
+	for _, trace := range traces {
+		filter := trace.Spec.Filter
+
+		id, present := trace.ObjectMeta.Labels[TRACE_ID]
+		if !present {
+			return errors.New("Trace has no ID!")
+		}
+
+		fmt.Fprintf(w, "%v\t%v\t%v\t%v\n",
+			filter.Namespace, filter.Podname, filter.ContainerName, id)
+	}
+
+	w.Flush()
+
+	return nil
+}
+
+// RunTraceAndPrintGenericOutput creates a trace, prints its output and deletes
+// it.
+// It equals calling separately CreateTrace(), then PrintTraceGenericOutput()
+// and DeleteTrace().
+// This function is thought to be used with "one-run" gadget, i.e. gadget
+// which runs a trace when it is created.
+func RunTraceAndPrintGenericOutput(config *TraceConfig, transformLine func(string) string) error {
+	traceID, err := CreateTrace(config)
+	if err != nil {
+		return fmt.Errorf("error creating trace: %w", err)
+	}
+
+	err = PrintTraceGenericOutput(traceID, config, transformLine)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%v\n", err)
+	}
+
+	return DeleteTrace(traceID)
+}
+
+// RunTraceAndPrintCustomOutput creates a trace, prints its output and deletes
+// it.
+// It equals calling separately CreateTrace(), then PrintTraceCustomOutput()
+// and DeleteTrace().
+// This function is thought to be used with "one-run" gadget, i.e. gadget
+// which runs a trace when it is created.
+func RunTraceAndPrintCustomOutput(config *TraceConfig, customResultsDisplay func(nodes *corev1.NodeList, results *gadgetv1alpha1.TraceList)) error {
```

Comment:
I know it is not strictly related to this PR but I think we can take the opportunity to remove the `nodes *corev1.NodeList` argument from the `customResultsDisplay()` function because it is not used. What do you think?

cc @alban